jung-visualization/src/test/java/edu/uci/ics/jung/visualization/spatial/SpatialGridTest.java

```diff
@@ -0,0 +1,40 @@
+package edu.uci.ics.jung.visualization.spatial;
+
+import java.awt.*;
+import java.util.Arrays;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Created by Tom Nelson. */
+public class SpatialGridTest {
+
+  Logger log = LoggerFactory.getLogger(SpatialGridTest.class);
+
+  @Test
+  public void testBoxLocations() {
+    SpatialGrid spatial = new SpatialGrid(new Rectangle(0, 0, 100, 100), 4, 4);
+    log.info("grid is " + spatial.getGrid());
```

Comment:
I admit I'm skeptical of the value of logging like this in tests.

I believe it would be better to assert on the behaviour that you expect `SpatialGrid` to exhibit, or even to query its internal values if needed (but this would be a kind of last resort IMO), so that the test fails if the `SpatialGrid` doesn't have the grid you expect.

Furthermore, I struggle to see the value of logging here where stdout would suffice. I'd appreciate any insight here.

Reply:
Drive by comment: if you guys decide to keep the logging, then write `log.info("grid is {}", spatial.getGrid());`. Idem elsewhere; leave string concatenation to the logging subsystem.

Reply:
Thanks! I am just now updating to log4j2 and changing the syntax as you noted.

Reply:
I agree that it is not really a unit test.